### PR TITLE
feat: Add moe load balancing metrics

### PR DESCRIPTION
**Pull request by yfw**
# What does this PR do ?

Port moe load balancing metrics from Megatron-LM to nemo-rl

# Issues
List issues that this PR closes ([syntax](https://docs.github.com/en/issues/tracking-your-work-with-issues/using-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)):

# Usage
* **You can potentially add a usage example below**

```python
# Add a code snippet demonstrating how to use this
```

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](/NVIDIA-NeMo/RL/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you run the unit tests and functional tests locally? Visit our [Testing Guide](/NVIDIA-NeMo/RL/blob/main/docs/testing.md) for how to run tests
- [ ] Did you add or update any necessary documentation? Visit our [Document Development Guide](/NVIDIA-NeMo/RL/blob/main/docs/documentation.md) for how to write, build and test the docs.

# Additional Information
* ...

## Release Notes

* **New Features**
  * Added MOE (Mixture of Experts) metrics tracking to training pipelines
  * Per-layer metrics logging available for detailed MOE diagnostics

* **Configuration**
  * Introduced `track_moe_metrics` and `moe_per_layer_logging` flags across training configurations (disabled by default)

* **Tests**
  * Added comprehensive unit tests for MOE metrics collection and aggregation

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: nemo_rl/models/megatron/common.py [hunk 1]
```diff
@@ -592,3 +597,49 @@ def broadcast_tensor(
...
+        total_loss_dict: If provided, accumulate means into this dict (by name).
+        per_layer_logging: If True, include per-layer values in the returned dict.
+
+    Returns:
+        dict[str, Any]: A flat dict of aggregated metrics. For each aux loss name,
+        the mean value is returned under the same key (e.g., "load_balancing_loss").
+        If per_layer_logging is True, per-layer values are returned under keys of the
+        form "moe/{name}_layer_{i}".
+    """
+    reduce_aux_losses_tracker_across_ranks()
```
Comment (terrykong) [inline nemo_rl/models/megatron/common.py:630]: 1. is this function destructive to the aux losses? is there any way to ensure it runs after forward-backward if so? if not possible, a comment is probably necessary
2. does the aux loss get computed in get_logprobs or other megatron policy worker methods? would the aux loss accumulate if not reset before the `train` method?

File: nemo_rl/models/policy/megatron_policy_worker.py [hunk 2]
```diff
@@ -1177,6 +1182,15 @@ def train(
             "all_mb_metrics": dict(mb_metrics),
             "grad_norm": torch.tensor([grad_norm]),
         }
+        # Collect MoE aux metrics averaged across microbatches
+        if self.cfg["megatron_cfg"]["track_moe_metrics"]:
```
Comment (terrykong) [inline nemo_rl/models/policy/megatron_policy_worker.py]: is there a reason not to track this? could we simply infer from model and always log to reduce number of flags?

File: nemo_rl/models/policy/megatron_policy_worker.py [hunk 3]
```diff
@@ -1177,6 +1182,15 @@ def train(
...
+            moe_loss_scale = 1.0 / max(1, total_num_microbatches)
+            moe_metrics = get_moe_metrics(
```
Comment (terrykong) [inline nemo_rl/models/policy/megatron_policy_worker.py:1187]: i recall there was an issue with CP or DP normalization and the aux loss. has that since been resolved?

Comment (yfw) [reply nemo_rl/models/policy/megatron_policy_worker.py:1187]: That fix hasn't been merged yet I believe CC @gshennvm

Comment (gshennvm) [reply nemo_rl/models/policy/megatron_policy_worker.py:1187]: this PR seems orthogonal, but the moe aux loss fix is still pending some tests

File: nemo_rl/models/megatron/common.py (hunk 1 above)
Comment (yfw) [reply nemo_rl/models/megatron/common.py:630]: 1. These aren't destructive to the actually aux losses used in training, we just keep a copy of the losses for tracking purposes (https://github.com/NVIDIA/Megatron-LM/blob/e8749f88691cac1eeefd11b6b68cb8a6557356d5/megatron/core/transformer/moe/moe_utils.py#L748)
2. The aux loss is only calculated during training and when grads are enabled (https://github.com/NVIDIA/Megatron-LM/blob/e8749f88691cac1eeefd11b6b68cb8a6557356d5/megatron/core/transformer/moe/router.py#L541), so for us it should only be in the `train` method.

File: nemo_rl/models/policy/megatron_policy_worker.py (hunk 2 above)
Comment (yfw) [reply nemo_rl/models/policy/megatron_policy_worker.py]: Made this change in https://github.com/NVIDIA-NeMo/RL/pull/1520/commits/bfaeb1c60f8810506a7a0e5606943fbe2f42c135
